src/coreComponents/finiteVolume/TwoPointFluxApproximation.cpp

```diff
@@ -109,7 +109,12 @@ void TwoPointFluxApproximation::computeCellStencil( MeshLevel & mesh ) const
   SortedArray< localIndex > regionFilter;
   for( string const & regionName : m_targetRegions.at( mesh.getParent().getParent().getName() ) )
   {
-    regionFilter.insert( elemManager.getRegions().getIndex( regionName ) );
+    ElementRegionBase const & region = elemManager.getRegion( regionName );
+    // Using the catalog name as a trick to select only cell element regions...
```

Comment:
I don't know why this is referred to as a "trick". There isn't much tricky about it. If we prefer, we can create a virtual function to return `typeid(T)` and compare that instead. Perhaps that would be less of a "trick"?

Reply:
Because in this TPFA code _which decides what are the connections between the discretized domains_,
we end up using features that are designed to be used 1) _for the user input file_ and 2) _the technical registration of products in our factories_.

The computation domain of the TPFA does not belong to the xml/factory world.
And vice versa.

Using/calling functions that are not of the same "_world_" or "_domain_" or "_level_" should not "_really_" happen.
Imagine that at some moment we change something about the `catalog`, TPFA gets impacted.

This is why I consider this is a trick.

PS: I'm pretty sure there is an established wording for this, but I cannot figure it out right now, sorry.
It will hopefully get back to my mind again 🤷

Reply:
In this case we might be able to rewrite this whole loop using our usual filtering technique (commentware, to be tested):
```c++
arrayView1d< string const > const targetRegions =
  m_targetRegions.at( mesh.getParent().getParent().getName() );

elemManager.forElementRegionsComplete< CellElementRegion >( targetRegions,
                                                            [&]( localIndex, 
                                                                 localIndex const ei, 
                                                                 CellElementRegion const & )
{
  regionFilter.insert( ei );
} );
```

In general, I agree that using strings as subtype discriminators is definitely a bit... unorthodox, even if there is no performance constraint. Luckily, due to our downcast-based filtering, we have largely avoided relying on catalog names for things like this. Not sure if this will be always possible though.